### PR TITLE
Fix/frontend issues

### PR DESCRIPTION
**Pull request by jbpenrath**
## Purpose

1. Old browsers does not support group regexp. So enrollment widget does not work on these browsers.
https://sentry.io/organizations/gip-fun-mooc/issues/2337189525/?environment=production&project=1454000&query=is%3Aunresolved

2. There are too much request to the search api endpoint.
From our search components, autocomplete and search queries are immediately fired on each user inputs that is too much. We have to debounce these requests. (The debounce was in place on SuggestField but did not work properly).

## Proposal

- [x] Remove use of group regexp from frontend
- [x] Debounce search requests

### CODE REVIEW

File: sandbox/settings.py [hunk 1]
```diff
@@ -543,9 +543,7 @@ class Base(StyleguideMixin, DRFMixin, RichieCoursesConfigurationMixin, Configura
 
     # For more details about CMS_CACHE_DURATION, see :
     # http://docs.django-cms.org/en/latest/reference/configuration.html#cms-cache-durations
-    CMS_CACHE_DURATIONS = values.DictValue(
-        {"menus": 3600, "content": 60, "permissions": 3600}
-    )
+    CMS_CACHE_DURATIONS = values.DictValue({"menus": 0, "content": 0, "permissions": 0})
```
Comment (sampaccoud) [inline sandbox/settings.py:546]: fail?

File: src/frontend/js/utils/api/lms/openedx-hawthorn.ts [hunk 2]
```diff
@@ -20,8 +20,10 @@ const API = (
   APIConf: AuthenticationBackend | LMSBackend,
   options?: ApiOptions,
 ): ApiImplementation => {
-  const extractCourseIdFromUrl = (url: string): Maybe<Nullable<string>> =>
-    url.match((APIConf as LMSBackend).course_regexp)?.groups?.course_id;
+  const extractCourseIdFromUrl = (url: string): Maybe<Nullable<string>> => {
+    const matches = url.match((APIConf as LMSBackend).course_regexp);
+    return matches && matches[1] ? matches[1] : null;
```
Comment (sampaccoud) [inline src/frontend/js/utils/api/lms/openedx-hawthorn.ts:25]: I'm wondering, do you see anyway to lint that this is not used in future code?

File: src/frontend/js/components/SearchSuggestField/index.tsx [hunk 3]
```diff
@@ -87,7 +87,10 @@ const SearchSuggestField = ({ context }: CommonDataProps) => {
       });
     }
   };
-  const updateCourseSearchParamsDebounced = debounce(searchAsTheUserTypes, 500, { maxWait: 1100 });
+  const updateCourseSearchParamsDebounced = useMemo(
+    () => debounce(searchAsTheUserTypes, 500, { maxWait: 1100 }),
+    [],
+  );
```
Comment (sampaccoud) [inline src/frontend/js/components/SearchSuggestField/index.tsx:93]: I don't understand how it is going to work. If you pass [] as params then they are never going to change and memoization is forever so debounce is called only once?

File: sandbox/settings.py (hunk 1 above)
Comment (jbpenrath) [reply sandbox/settings.py:546]: Oups !

File: src/frontend/js/components/SearchSuggestField/index.tsx (hunk 3 above)
Comment (jbpenrath) [reply src/frontend/js/components/SearchSuggestField/index.tsx:93]: Yes, the debounce method is called once, so only once timer is instantiated during the component life cycle. Previously, we called debounce at every render, so a timer was created each time and debounce cannot work.

Comment (sampaccoud) [reply src/frontend/js/components/SearchSuggestField/index.tsx:93]: Would be nice if this can be tested ^^

Comment (jbpenrath) [reply src/frontend/js/components/SearchSuggestField/index.tsx:93]: Yes I'm currently fighting to fix tests and remove this TODO ^^.
https://github.com/openfun/richie/blob/f2c68abe06d2e711f5bea0c9835f84de82777a81/src/frontend/js/components/SearchSuggestField/index.spec.tsx#L470

File: src/frontend/js/utils/api/lms/openedx-hawthorn.ts (hunk 2 above)
Comment (jbpenrath) [reply src/frontend/js/utils/api/lms/openedx-hawthorn.ts:25]: For this case, as the regex is provided by the backend it is something hard to detect from the frontend.

Comment (sampaccoud) [reply src/frontend/js/utils/api/lms/openedx-hawthorn.ts:25]: ok... You could look for `.groups` but that may be a bit too common...

Comment (jbpenrath) [reply src/frontend/js/utils/api/lms/openedx-hawthorn.ts:25]: The error occured when the regex is instantiate. We couldn't even test whether the `groups` attribute exists. We could try to detect if the regex course string contains `<?.*>` but from my point of view it is not a good idea to add this complexity to support old browsers.

Comment (sampaccoud) [reply src/frontend/js/utils/api/lms/openedx-hawthorn.ts:25]: :+1: I follow you.

File: src/frontend/js/components/SearchSuggestField/index.tsx [hunk 4]
```diff
@@ -66,28 +66,39 @@ const SearchSuggestField = ({ context }: CommonDataProps) => {
    * Helper to update the course search params when the user types. We needed to take it out of
    * the `onChange` handler to wrap it in a `debounce` (and therefore a `useRef` to make the
    * debouncing effective).
+   *
+   * This method should be memoized and updated only courseSearchParams changed.
```
Comment (sampaccoud) [inline src/frontend/js/components/SearchSuggestField/index.tsx:70]:
```suggestion
   * This method should be memoized and updated only when courseSearchParams change.
```

**Review by sampaccoud — APPROVED** [review]

**Pull request merged**